testsuites/listener/shared/client_sg/test_replication.py

```diff
@@ -139,6 +141,106 @@ def test_initial_pull_replication(setup_client_syncgateway_test, continuous):
 @pytest.mark.listener
 @pytest.mark.syncgateway
 @pytest.mark.replication
+@pytest.mark.session
+@pytest.mark.parametrize("num_docs, need_attachments, replication_after_backgroundApp", [
+    (1000, True, False),
+    (10000, False, False),
+    (10000, False, False),
+    (100000, False, False),
+    (100000, False, False),
+    (100000, False, True)
+])
+def test_initial_pull_replication_background_apprun(setup_client_syncgateway_test, num_docs, need_attachments, replication_after_backgroundApp):
+    """
+    1. Prepare sync-gateway to have 10000 documents.
+    2. Create a single shot / continuous pull replicator and to pull the docs into a database.
+    3. Verify if all of the docs get pulled.
+    Referenced issue: couchbase/couchbase-lite-android#955.
+    """
+
+    sg_db = "db"
+    ls_db = "ls_db"
+
+    cluster_config = setup_client_syncgateway_test["cluster_config"]
+    sg_mode = setup_client_syncgateway_test["sg_mode"]
+    ls_url = setup_client_syncgateway_test["ls_url"]
+    sg_one_admin = setup_client_syncgateway_test["sg_admin_url"]
+    sg_one_public = setup_client_syncgateway_test["sg_url"]
+    liteserv = setup_client_syncgateway_test["liteserv"]
+    liteserv_platform = setup_client_syncgateway_test["liteserv_platform"]
+
+    sg_config = sync_gateway_config_path_for_mode("listener_tests/listener_tests", sg_mode)
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+    
+    if liteserv_platform != "ios":
+        pytest.skip('This test only valid for mobile')
```

Comment:
ios is also a mobile platform. We can be specific about platform or instead we can use that "test is not valid for ios".

Reply:
modified